raiden/tests/unit/migrations/test_v16_to_v17.py

```diff
@@ -1,32 +1,27 @@
-import random
+import json
 from datetime import datetime
 from pathlib import Path
 from unittest.mock import patch
 
-from raiden.storage.serialize import JSONSerializer
-from raiden.storage.sqlite import SerializedSQLiteStorage, SQLiteStorage
-from raiden.tests.utils import factories
-from raiden.transfer.state_change import ActionInitChain
+from raiden.storage.sqlite import SQLiteStorage
+from raiden.tests.utils.migrations import create_fake_web3_for_block_hash
 from raiden.utils.upgrades import UpgradeManager
 
 
 def setup_storage(db_path):
-    storage = SerializedSQLiteStorage(str(db_path), JSONSerializer())
-
-    chain_state_data = Path(__file__).parent / 'data/v16_chainstate.json'
-    chain_state = chain_state_data.read_text()
-
+    # For a raw ActionInitChain let's get the v18 one. It should be the same as v16
```

Comment:
why? i would keep each migration test strictly separated from other migrations so that it is tested properly. Unless you have a strong reason for this.

Reply:
They are separate. This is just the data of the action init chain that was generated and is identical for v18 and v16. Makes no sense for me to go and check out a commit of v16 DB, run a test, generate new json for a v16 `ActionInitChain` and commit it for tests when it's identical.

Reply:
Something is really wrong here, we can't have a situation were we have to keep going back to upgrade old tests

Reply:
> Something is really wrong here, we can't have a situation were we have to keep going back to upgrade old tests

That is true. The reason for this is that all these tests run the upgrades till the end. So the v16->v17 test actually does:

- v16->v16
- v17->v18
- v18->v19

This is why I made some tiny changes to the old tests and wanted to ask @rakanalh if this was intended test behavior.

But yeah this is not sustainable and probably tests should just be limited to their own upgrade and not let run till latest.

Reply:
okay, so lets fix the test and run only the migration of interest instead of everything

Reply:
agreed

Reply:
This should do it: https://github.com/raiden-network/raiden/pull/3550/commits/3b8b60ebca1482ceb7ddaea1a8bcc828c371f17f